### ISSUE TITLE
arrow v11 doesnt build on mac M1

### ISSUE THREAD

**Issue by madisonchamberlain** [snowflakedb/gosnowflake#789]
Please answer these questions before submitting your issue. 
In order to accurately debug the issue this information is required. Thanks!

1. What version of GO driver are you using?
latest
   
2. What operating system and processor architecture are you using?
Mac m1
   
3. What version of GO are you using?
run `go version` in your console
1.20

4.Server version:* E.g. 1.90.1
You may get the server version by running a query:
```
SELECT CURRENT_VERSION();
```
5. What did you do?
run go build on mac M1

6. What did you expect to see?
```
➜ go build ./...
# github.com/apache/arrow/go/v11/arrow/compute/internal/kernels
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:816:10: undefined: getArithmeticOpIntegral
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:818:10: undefined: getArithmeticOpIntegral
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:820:10: undefined: getArithmeticOpIntegral
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:822:10: undefined: getArithmeticOpIntegral
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:824:10: undefined: getArithmeticOpIntegral
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:826:10: undefined: getArithmeticOpIntegral
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:828:10: undefined: getArithmeticOpIntegral
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:830:10: undefined: getArithmeticOpIntegral
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:832:10: undefined: getArithmeticOpFloating
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:834:10: undefined: getArithmeticOpFloating
../../../go/pkg/mod/github.com/apache/arrow/go/v11@v11.0.0/arrow/compute/internal/kernels/base_arithmetic.go:834:10: too many errors
```
   What should have happened and what happened instead?
Nothing
8. Can you set logging to DEBUG and collect the logs?

   https://community.snowflake.com/s/article/How-to-generate-log-file-on-Snowflake-connectors
   
9. What is your Snowflake account identifier, if any? (Optional)

**Comment by sfc-gh-dszmolka** [comment]
hello and thank you for raising this issue. it seems to be coming from an upstream dependency Apache Arrow, and seems to be related to and Issue opened against this dependency:
* https://github.com/apache/arrow/issues/34839 (Issue)
* https://github.com/apache/arrow/pull/34840 (PR, merged April 3)

the PR seems to be already part of [Arrow version 12.0.0](https://github.com/apache/arrow/blob/go/v12.0.0/go/arrow/compute/internal/kernels/basic_arithmetic_noasm.go#L17)

but currently we seem to use [Arrow v10.0.1](https://github.com/snowflakedb/gosnowflake/blob/v1.6.20/go.mod#L9) and even though we did [bump Arrow to v11](https://github.com/snowflakedb/gosnowflake/commit/77919ad72dd40a483b06cc90ec396be97db2dffc#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6) since the change is only available for around a month and only part of Arrow 12, the gosnowflake library doesn't yet have it. we'll look into it how this improvement could be scheduled.

for now as a workaround could you please try building with `-tags=noasm` as suggested in Arrow issue 34839 and see if it helps you ?

**Comment by sfc-gh-dszmolka** [comment]
#790

**Comment by madisonchamberlain** [comment]
That was a fast fix! Thank you!

**Comment by sfc-gh-dszmolka** [comment]
(going to reopen this until the May release is out, just for administrative purposes and tracking)

**Comment by sfc-gh-dszmolka** [comment]
fix released with version [1.6.21](https://github.com/snowflakedb/gosnowflake/releases/tag/v1.6.21), closing this issue
